src/utils/getRegexp.ts

```diff
@@ -0,0 +1,101 @@
+import {FileEntry} from '@models/fileentry';
+
+export type MatchParamProps = {
+  matchCase: boolean;
+  matchWholeWord: boolean;
+  regExp: boolean;
+};
+
+export function getRegexp(query: string, params: MatchParamProps): RegExp {
+  let matchParams = 'gi'; // global, case insensitive by default
+  if (params.matchCase) {
+    matchParams = 'g';
+  }
+  if (!params.regExp) {
+    query = query.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&');
+  }
+
+  let queryRegExp = new RegExp(query, matchParams);
+
+  if (params.matchWholeWord) {
+    queryRegExp = new RegExp(`\\b${query}\\b`, matchParams);
+  }
+  if (params.regExp) {
+    queryRegExp = new RegExp(query, matchParams);
+  }
+
+  return queryRegExp;
+}
+
+type Props = {
```

Comment:
this looks similar to the MatchNode model, maybe it's not needed?
on the `decorateMatch` function below you could do:

```ts
function decorateMatch(text: string, query: string, fromIndex = 0): Omit<MatchNode, 'textWithHighlights'> {
```

Reply:
fixed